### PR TITLE
SW-4963: Add the approve deliverable confirmation screen

### PR DESCRIPTION
**Pull request by joshbuchea**
This PR includes changes to support showing a confirmation dialog when approving a deliverable.

## Example

localhost_3000_accelerator_deliverables_1 (5)

### CODE REVIEW

File: src/scenes/AcceleratorRouter/ApproveDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { Button, DialogBox } from '@terraware/web-components';
+
+import strings from 'src/strings';
+
+export type ApproveDialogProps = {
+  onClose: () => void;
+  onSubmit: () => void;
+};
+
+export default function ApproveDialog({ onClose, onSubmit }: ApproveDialogProps): JSX.Element {
```
Comment (karthikbtf) [inline src/scenes/AcceleratorRouter/ApproveDialog.tsx:11]: ApproveDeliverableDialog? And the filename.

File: src/scenes/AcceleratorRouter/ApproveDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const theme = useTheme();
+
+  const approve = () => {
+    onSubmit();
```
Comment (karthikbtf) [inline src/scenes/AcceleratorRouter/ApproveDialog.tsx:15]: onSubmit also closes the dialog, we don't need the onClose here.

File: src/scenes/AcceleratorRouter/ApproveDialog.tsx (hunk 1 above)
Comment (karthikbtf) [inline src/scenes/AcceleratorRouter/ApproveDialog.tsx:11]: This looks like a good candidate for generic confirm dialog component, I created a [ticket](https://terraformation.atlassian.net/browse/SW-4969).

File: src/scenes/AcceleratorRouter/DeliverableViewWrapper.tsx [hunk 3]
```diff
@@ -10,30 +10,36 @@ import OptionsMenu from 'src/components/common/OptionsMenu';
...
   const { deliverable } = useFetchDeliverable({ deliverableId: Number(deliverableId) });
 
-  // temporary solution until we have the confirmation modal design
   const setStatus = useCallback(
     (status: DeliverableStatusType) => {
       if (deliverable?.id !== undefined) {
-        if (window.confirm(`Are you sure you want to set the status to ${status} ?`)) {
-          update({ ...deliverable, status });
-        }
+        update({ ...deliverable, status });
```
Comment (karthikbtf) [inline src/scenes/AcceleratorRouter/DeliverableViewWrapper.tsx:30]: conflicts with Nick's PR just a heads-up

**Review by karthikbtf — APPROVED** [review]
looks good, a few comments

File: src/scenes/AcceleratorRouter/ApproveDialog.tsx (hunk 1 above)
Comment (joshbuchea) [reply src/scenes/AcceleratorRouter/ApproveDialog.tsx:11]: Good call 👍

**Review by karthikbtf — APPROVED** [review]
🙌

**Pull request merged**